pkg/cloud/azure/services/virtualmachines/virtualmachines.go

```diff
@@ -98,11 +99,20 @@ func (s *Service) CreateOrUpdate(ctx context.Context, spec azure.Spec) error {
 		return errors.Wrapf(err, "failed to generate random string")
 	}
 
+	var zones *[]string
+	if vmSpec.Zone != "" {
+		zones = new([]string)
+		*zones = append(*zones, vmSpec.Zone)
+	} else {
+		zones = nil
+	}
+
 	future, err := s.Client.CreateOrUpdate(
 		ctx,
 		s.Scope.ClusterConfig.ResourceGroup,
 		vmSpec.Name,
 		compute.VirtualMachine{
+			Zones:    zones,
```

Comment:
if we declares `zones` as a slice instead of a pointer to slice, we'd just need to do this. 
```suggestion
			Zones:    &zones,
```

Reply:
Azure doesnt like it, it errors out saying no zones specified, you have to either pass nil or some valid value. Its the limitation of azure golang sdk api.